olp-cpp-sdk-core/include/olp/core/network2/NetworkSettings.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <olp/core/CoreApi.h>
+
+#include "NetworkProxySettings.h"
+
+namespace olp {
+namespace network2 {
+
+/**
+ * @brief The NetworkSettings class contains configuration for the network
+ */
+class CORE_API NetworkSettings {
```

Comment:
Should the _NetworkSettings_ class be subclassed by the user? If yes, provide virtual dtor. If not - better to make it final.